festim/concentration/traps/traps.py

```diff
@@ -72,7 +72,16 @@ def define_variational_problem_extrinsic_traps(self, dx, dt, T):
     def solve_extrinsic_traps(self):
         for trap in self.traps:
             if isinstance(trap, festim.ExtrinsicTrapBase):
-                f.solve(trap.form_density == 0, trap.density[0], [])
+                du_t = f.TrialFunction(trap.density[0].function_space())
+                J_t = f.derivative(trap.form_density, trap.density[0], du_t)
+                problem = f.NonlinearVariationalProblem(
+                    trap.form_density, trap.density[0], [], J_t)
```

Comment:
I wonder if the Jacobian needs to be explicitly defined here. Wouldn't `problem = f.NonlinearVariationalProblem(trap.form_density, trap.density[0], [])` be enough?

Reply:
Yeah thats true its not likely needed, but I'll do some quick testing to see if it makes a difference

Reply:
I think the default behaviour of fenics is to do exactly what lines 75-76 do. So it would just be saving 2 lines :-)